schema/Retail/RET/on_search.ts

```diff
@@ -619,10 +718,67 @@ export const onSearchSchema = {
                               },
                             },
                             required: ['code', 'value'],
+                            additionalProperties: false,
+                            allOf: [
+                              {
+                                if: {
+                                  properties: {
+                                    code: {
+                                      const: 'type_validity',
+                                    },
+                                  },
+                                },
+                                then: {
+                                  properties: {
+                                    value: {
+                                      format: 'duration',
+                                      errorMessage: 'Duration must be RFC3339 duration.',
+                                    },
+                                  },
+                                },
+                              },
+                              {
+                                if: {
+                                  properties: {
+                                    code: {
+                                      const: 'last_update',
+                                    },
+                                  },
+                                },
+                                then: {
+                                  properties: {
+                                    value: {
+                                      description: 'RFC3339 UTC timestamp format',
+                                      format: 'date-time',
+                                      errorMessage: 'Time must be RFC3339 UTC timestamp format.',
+                                    },
+                                  },
+                                },
+                              },
+                              {
+                                if: {
+                                  properties: {
+                                    code: {
+                                      const: 'min_value',
+                                    },
+                                  },
+                                },
+                                then: {
+                                  properties: {
+                                    value: {
+                                      type: 'string',
+                                      pattern: '^\\d+(\\.\\d{1,2})?$',
+                                      errorMessage: 'Amount must be a number with up to 2 decimal places.',
```

Comment:
Replace the error message as: amount must be in stringified number format with up to 2 decimal places.